classes/output/core_renderer.php

```diff
@@ -131,6 +131,170 @@ public function main_menu() {
         $this->page->requires->js_call_amd('theme_wwu2019/menu', 'init');
         return $this->render_from_template('theme_wwu2019/menu', $templatecontext);
     }
+    /**
+     * Add additional items to edit and to go to bottom of page.
+     * Allow plugins to provide some content to be rendered in the navbar.
+     * The plugin must define a PLUGIN_render_navbar_output function that returns
+     * the HTML they wish to add to the navbar.
+     *
+     * @return string HTML for the navbar
+     */
+    public function navbar_plugin_output() {
+        $output = '';
+        if ($this->page->user_allowed_editing()) {
+            $output .= $this->add_edit_button();
+        }
+        if (!$this->is_login_page()) {
+            $output .= $this->add_endofpage_button();
+        }
+        // Give subsystems an opportunity to inject extra html content. The callback
+        // must always return a string containing valid html.
+        foreach (\core_component::get_core_subsystems() as $name => $path) {
+            if ($path) {
+                $output .= component_callback($name, 'render_navbar_output', [$this], '');
+            }
+        }
+
+        if ($pluginsfunction = get_plugins_with_function('render_navbar_output')) {
+            foreach ($pluginsfunction as $plugintype => $plugins) {
+                foreach ($plugins as $pluginfunction) {
+                    $output .= $pluginfunction($this);
+                }
+            }
+        }
+        return $output;
+    }
+
+    /**
+     * Renders an edit button.
+     * @return string HTML for editbutton
+     */
+    private function add_edit_button() {
+        $html = '';
+        $url = $this->switch_pagetype();
+        $url->param('sesskey', sesskey());
+
+        if ($this->page->user_is_editing()) {
+            $icon = 'power-off';
+        } else {
+            $icon = 'edit';
+        }
+        $attributes['class'] = 'icon fa fa-' . $icon;
+        $attributes['aria-hidden'] = 'true';
+        $attributes['href'] = $url;
+        $tag = \html_writer::tag('a', '', $attributes);
+        $html .= \html_writer::div($tag, 'popover-region-toggle nav-link');
+        return $html;
+    }
+
+    /**
+     * Checks the current pagetype and sets the edit param based on the pagetype.
+     * TODO: n_herr03 the switch statement is not exhaustive. Moreover, not all cases are tested.
+     * @return object url with the required edit options
+     */
+    private function switch_pagetype() {
```

Comment:
Can we use a more speaking name, such as `get_edit_button_url_by_pagetype`?

Also, the switch is exhaustive: There is a `default` clause at the end.

Reply:
https://github.com/learnweb/moodle-theme_wwu2019/commit/8bb0ad82a8a13a84b8da70d60f5102055a5d2ee2 and https://github.com/learnweb/moodle-theme_wwu2019/commit/f03d30f29764baebdbc7925fb99fea3bea57a721